### DISCUSSION TITLE
Jittering fixation durations

### DISCUSSION THREAD

**Discussion by ssyoon** [jspsych/jsPsych#829] [Q&A] [answered]
Hi, 

I'm wondering if there is a way to jitter fixation durations randomly. 
For example, instead of using a specific fixed duration for 'trial_duration' (e.g., 2000 ms), I'm wondering if I can change the duration so that 'trial_duration' can be between 2-4 seconds that vary randomly trial-by-trial. 
If you could share any ideas about this, it would be appreciated. 
Thank you for your help in advance. 

Best, 
Sangsuk

**Comment by becky-gilbert** [comment, marked as answer]
Yes, you can do this using a function for the fixation's duration parameter. There's one example here: https://www.jspsych.org/tutorials/rt-task/#part-7-using-functions-to-generate-parameters

And here's another example for writing a function that will randomize between 2 and 4 seconds:
```javascript
var fixation = {
  ...
  trial_duration: function() {
    var rand_iti = Math.floor(Math.random() * (4000 - 2000)) + 2000;
    return rand_iti;
}
```

**Reply by ssyoon** [reply]
Thank you so much for your answer! 
This works perfectly!
